src/Query/Builder.php

```diff
@@ -1071,7 +1071,14 @@ protected function compileWheres(): array
             }
 
             // Merge the compiled where with the others.
-            $compiled = array_merge_recursive($compiled, $result);
+            // array_merge_recursive can't be used here because it converts int keys to sequential int.
```

Comment:
`array_merge_recursive` get any int-key as a list and reset the value to `0`.

The merge logic have to be fine-coded to only merge lists of logical operators (`$and`, `$or`, `$nor`).